### PR TITLE
feat: upgrade builtin types

### PR DESCRIPTION
**Pull request by togami2864**
**Description:**
This PR upgrades built in types in `./crates//stc_ts_builtin_types/lib/*` and enable Es2021 & Es2022.
Got `d.ts` files from [TypeScript main branch(2023/01/20)](https://github.com/microsoft/TypeScript).

**BREAKING CHANGE:**

**Related issue (if exists):**
#467
@kdy1  This PR makes 4 extra panics and I did't enable  `es2022.intl.d.ts` because it causes a lot of regression.
Since the diff is so large, can I split the PR with another issue?

### CODE REVIEW

**Comment by kdy1** [comment]
Of course, splitting PR is fine

File: crates/stc_ts_builtin_types/lib/es2022.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/// <reference lib="es2021" />
+/// <reference lib="es2022.array" />
+/// <reference lib="es2022.error" />
+// /// <reference lib="es2022.intl" />
```
Comment (togami2864) [inline crates/stc_ts_builtin_types/lib/es2022.d.ts:4]: Temporarily disabled `<reference lib="es2022.intl" />`

File: crates/stc_ts_file_analyzer/tests/pass/types/conditional/conditionalTypes1/3.swc-stderr [hunk 2]
```diff
@@ -6,7 +6,8 @@
    `----
 
 Error: 
-  > T extends (null | undefined) ? never : T
+  > (T & {
+  | })
```
Comment (togami2864) [inline crates/stc_ts_file_analyzer/tests/pass/types/conditional/conditionalTypes1/3.swc-stderr:10]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-8.html

File: crates/stc_ts_type_checker/tests/tsc.rs [hunk 3]
```diff
@@ -891,7 +921,9 @@ fn libs_with_deps(libs: &[Lib]) -> Vec<Lib> {
             | Lib::Scripthost
             | Lib::WebworkerGenerated
             | Lib::WebworkerImportscripts
-            | Lib::WebworkerIterableGenerated => {}
+            | Lib::WebworkerIterableGenerated
+            | Lib::Decorators
+            | Lib::DecoratorsLegacy => {}
```
Comment (togami2864) [inline crates/stc_ts_type_checker/tests/tsc.rs:926]: I didn't understand where to write these.
Referenced by [es5.d.ts](https://github.com/microsoft/TypeScript/blob/83069ef10d5b9c8686f0a821955dd3fd7b6119a0/src/lib/es5.d.ts#L1-L2)

File: crates/stc_ts_file_analyzer/tests/pass/types/conditional/conditionalTypes1/3.swc-stderr (hunk 2 above)
Comment (kdy1) [reply crates/stc_ts_file_analyzer/tests/pass/types/conditional/conditionalTypes1/3.swc-stderr:10]: Yeah I know the reason

File: crates/stc_ts_type_checker/tests/tsc.rs (hunk 3 above)
Comment (kdy1) [reply crates/stc_ts_type_checker/tests/tsc.rs:926]: I think it's fine to figure it out in follow-up PRs

**Review by kdy1 — DISMISSED** [review]
Thank you!

**Comment by kdy1** [comment]
Oh CI failed

**Comment by kdy1** [comment]
You can simply remove the test

**Review by kdy1 — APPROVED** [review]

**Pull request merged**